src/main/java/org/folio/service/InvoiceLineStorageService.java

```diff
@@ -0,0 +1,70 @@
+package org.folio.service;
+
+import static org.folio.rest.impl.InvoiceStorageImpl.INVOICE_LINES_PREFIX;
+import static org.folio.rest.utils.ResponseUtils.buildBadRequestResponse;
+import static org.folio.rest.utils.ResponseUtils.buildErrorResponse;
+import static org.folio.rest.utils.ResponseUtils.buildNoContentResponse;
+import static org.folio.rest.utils.ResponseUtils.buildResponseWithLocation;
+import static org.folio.rest.utils.RestConstants.OKAPI_URL;
+
+import javax.ws.rs.core.Response;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.folio.dao.lines.InvoiceLinesDAO;
+import org.folio.rest.jaxrs.model.InvoiceLine;
+import org.folio.rest.jaxrs.model.InvoiceLineAuditEvent;
+import org.folio.rest.persist.DBClient;
+import org.folio.service.audit.AuditOutboxService;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@RequiredArgsConstructor
+public class InvoiceLineStorageService {
+
+  private final InvoiceLinesDAO invoiceLinesDAO;
+  private final AuditOutboxService auditOutboxService;
+
+  public void createInvoiceLine(InvoiceLine invoiceLine, Handler<AsyncResult<Response>> asyncResultHandler,
+                                Context vertxContext, Map<String, String> headers) {
+    log.info("createInvoiceLine:: Creating a new invoiceLine by id: {}", invoiceLine.getId());
+    new DBClient(vertxContext, headers).getPgClient()
+      .withTrans(conn -> invoiceLinesDAO.createInvoiceLine(invoiceLine, conn)
+        .compose(invoiceLineId -> auditOutboxService.saveInvoiceLineOutboxLog(conn, invoiceLine, InvoiceLineAuditEvent.Action.CREATE, headers))
+        .compose(v -> auditOutboxService.processOutboxEventLogs(headers, vertxContext)))
```

Comment:
we should invoke auditOutboxService.processOutboxEventLogs(headers, vertxContext)) when transaction is completed.
It means in onComplete section

Reply:
Thanks, should be correct now